cw/messaging.py

```diff
@@ -1,19 +1,26 @@
 import pika
 
 
-def open_connection (cfg):
-    return pika.BlockingConnection(
-        pika.ConnectionParameters(host=cfg['rabbit-host'],
-                                  port=cfg['rabbit-port'],
-                                  ssl=cfg['rabbit-ssl']))
+class MessagingException(Exception):
+    pass
+
+
+def open_connection(cfg):
+    try:
+        return pika.BlockingConnection(
+            pika.ConnectionParameters(host=cfg['rabbit-host'],
+                                      port=cfg['rabbit-port'],
+                                      ssl=cfg['rabbit-ssl']))
+    except Exception as e:
+        raise MessagingException("problem establishing rabbitmq connection: {}".format(e))
 
 
 def close_connection(conn):
     if conn is not None and conn.is_open:
         try:
             conn.close()
         except Exception as e:
-            pass
+            raise MessagingException("Problem closing rabbitmq connection: {}".format(e))
```

Comment:
raising exception from closing connection probably not what we want as it wouldn't necessarily stop functioning of code.  Better to log only here (definitely log this error regardless) or raise a specialized exception describing the situ and then ignoring further up the call stack.